### DISCUSSION TITLE
Safe environment to run the project

### DISCUSSION THREAD

**Discussion by OscarGA** [mtxr/vscode-sqltools#1213] [Show and tell]
Hi community,
I was dealing with some issues in my local PC trying to setup the environment to run this project, so I ended up with a creation of a simple Docker file that I created to setup and isolate the environment.
Since I'm new to typescript development, this was my solution to all the problems I had while trying to set up the environment to start contributing.

you can find the link to the Docker file [here](https://github.com/OscarGA/vscode-sqltools-dockerfile/blob/b3ead1aae054067bdcf26f4d72d6da8e75231bf3/Dockerfile).

**Comment by richb-hanover** [comment]
@OscarGA I am a huge fan of Docker for dev environments. (I too have suffered from wrangling the versions of all the tools into alignment...)

But... You've left off one bit information that would double the value of your Dockerfile.

Would you consider adding a set of instructions that describe how to invoke it? It could go on the top of the Dockerfile, and I'm guessing that it's as simple as:

```
cd <directory that has the SQLtools files>
docker run ...all the options...
```
(I am sure I could farble around and get it working, but you already know this information. Placing those instructions in the Dockerfile would save me (and every other newcomer) minutes (or hours?) getting it going). Thanks.

**Reply by gjsjohnmurray** [reply]
The project is already equipped to leverage this extension

https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers

See https://code.visualstudio.com/docs/devcontainers/containers for the VS Code documentation about this way of doing development in a container.

Feedback is welcome, and we should probably get info added to our docs at https://vscode-sqltools.mteixeira.dev/en/contributing/building.html.

**Comment by richb-hanover** [comment]
Better and better! I am just seeing the _.devcontainer_ directory now. I'm just a consumer of the SQLTools extension, but it's comforting to know that if I had a hankering to do real software development, I would have a safe environment for it. 

Are there instructions for telling people like @OscarGA how to use it after opening it in VS Code? Thanks again for a really useful project.

**Comment by gjsjohnmurray** [comment]
@OscarGA you may like to try the following steps (which assume you have installed Docker Desktop):

1. If you don't already have a fork of the repo, create one.
2. Open VS Code on your desktop. If a folder or workspace reopens there automatically, close it.
3. From Command Palette choose `Dev Containers: Clone Repository in Container Volume...` and respond as follows:

- `Enter` on first prompt (or click the dropdown option) to clone from GitHub.
- Type enough to locate your fork, then pick it.
- Pick the `dev` branch.

4. Optionally, click the notification link to watch log messages from container creation.
5. When the devcontainer starts, ignore the "Open Workspace" notification which appears in the bottom right of VS Code. This action needs to wait until the postCreateCommand steps have completed.
6. When postStartCommand has eventually completed (be patient), use notification button to open `vscode-sqltools.code-workspace`. If the notification has disappeared, click on the bell in the status bar to bring it back.
7. If a notification advises you that devcontainer.json has changed, click "Ignore".

If you plan to make changes and submit a PR, create a new branch first.

Feedback on these steps is welcome, and when they're verified/improved we could add them into the docs, for example at https://vscode-sqltools.mteixeira.dev/en/contributing/building. In fact, you could edit src/pages/en/contributing/building.mdx on a branch and submit the changes as a PR.
